File: aoc2023/src/py/01.py
import sys

res = 0
ll = {'0':0,'1':1,'2':2,'3':3,'4':4,'5':5,'6':6,'7':7,'8':8,'9':9,
'one': 1, 'two': 2, 'three': 3, 'four': 4, 'five':5, 'six':6, 'seven':7, 'eight':8, 'nine':9}
for line in sys.stdin:
    l = []
    if not line:
        pass
    else:
        for i in range(0, len(line)):
            for j in ll:
                if line[i:i+len(j)] == j:
                    l.append(j)
                    break

    res += int(str(ll[l[0]]) + str(ll[l[-1]]))

print(res)

File: aoc2023/src/bin/01.rs
#![allow(unused_must_use)]
#[allow(unused_imports)]
#[allow(unused_variables)]
use std::cmp::{max, min};

use aoc2023::UnsafeScanner;
use std::io::{stdin, stdout, BufWriter, Write};

pub fn main() {
    let mut scan = UnsafeScanner::new(stdin().lock());
    let mut out = BufWriter::new(stdout().lock());

    let d = [
        "one", "two", "three", "four", "five", "six", "seven", "eight", "nine",
    ];

    let mut res = 0;
    while let Some(line) = scan.read_line() {
        let line = line.as_bytes();
        let mut sum = 0;
        for (i, c) in line.iter().enumerate() {
            let c = *c as usize - '0' as usize;
            if c > 0 && c <= 9 {
                if sum == 0 {
                    sum = 11 * c;
                } else {
                    sum = sum / 10 * 10 + c;
                }
            } else {
                for (index, num) in d.iter().enumerate() {
                    if i + num.len() <= line.len() && &line[i..i + num.len()] == num.as_bytes() {
                        if sum == 0 {
                            sum = 11 * (index + 1);
                        } else {
                            sum = sum / 10 * 10 + index + 1;
                        }
                    }
                }
            }
        }
        res += sum;
    }
    writeln!(out, "{}", res);
}
